components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMgtDBUtil.java

```diff
@@ -222,4 +227,48 @@ public static void rollbackConnection(Connection connection, String error, SQLEx
             throw e;
         }
     }
+
+    /**
+     * Handle connection rollback logic. Rethrow original exception so that it can be handled centrally.
+     * @param rs result set
+     * @throws SQLException sql exception
+     * @throws APIManagementException api management exception
+     */
+    public static List<APIRevisionDeployment> mergeRevisionDeploymentDTOs(ResultSet rs) throws APIManagementException,
+            SQLException {
+        List<APIRevisionDeployment> apiRevisionDeploymentList = new ArrayList<>();
+        Map<String, APIRevisionDeployment> uniqueSet = new HashMap<>();
+        while (rs.next()) {
+            APIRevisionDeployment apiRevisionDeployment;
+            String environmentName = rs.getString("NAME");
+            String vhost = VHostUtils.resolveIfNullToDefaultVhost(environmentName,
+                    rs.getString("VHOST"));
+            String revisionUuid = rs.getString("REVISION_UUID");
+            String uniqueKey = (environmentName != null ? environmentName : "") +
+                    (vhost != null ? vhost : "") + (revisionUuid != null ? revisionUuid : "");
+            if (!uniqueSet.containsKey(uniqueKey)) {
+                apiRevisionDeployment = new APIRevisionDeployment();
+                apiRevisionDeployment.setDeployment(environmentName);
+                apiRevisionDeployment.setVhost(vhost);
+                apiRevisionDeployment.setRevisionUUID(revisionUuid);
+                apiRevisionDeployment.setDisplayOnDevportal(rs.getBoolean("DISPLAY_ON_DEVPORTAL"));
+                apiRevisionDeployment.setDeployedTime(rs.getString("DEPLOY_TIME"));
+                apiRevisionDeployment.setSuccessDeployedTime(rs.getString("DEPLOYED_TIME"));
```

Comment:
Could you pls do a quick sample dump of those rs.getString("DEPLOY_TIME") and rs.getString("DEPLOYED_TIME") here? need to check whether we are getting the timezone with this

Reply:
getString() provide the value in this format 2021-08-24 10:56:18.391. We are adding the timestamp to DB at insert time in UTC

Reply:
this will be further investigated in this issue https://github.com/wso2-enterprise/choreo/issues/7784